### PR TITLE
Improve searching using the Find command

### LINKED ISSUE #61: Improve find feature for v1.2b
- expand find command to search beyond name into other fields

### PR DESCRIPTION
**Pull request by gowribhat**
- Change the find command to be able to search for all fields of a student
- Can also search with partial keywords
- Add test cases

closes #61

### CODE REVIEW

File: src/test/java/seedu/address/model/student/ClassGroupTest.java [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
     private final String nonEmptyClassGroup = "Some class group.";
     private ClassGroup classGroup = new ClassGroup(nonEmptyClassGroup);
 
+    @Test
+    void contains_classGroupContainsKeyword_returnsTrue() {
+        // Blank keyword
+        assertTrue(new ClassGroup("cs2030 lab 31").contains(""));
+
+        // Exact keyword
+        assertTrue(new ClassGroup("cs2030 lab 31").contains("cs2030 lab 3"));
```
Comment (thamruicong) [inline src/test/java/seedu/address/model/student/ClassGroupTest.java]:
```suggestion
        assertTrue(new ClassGroup("cs2030 lab 31").contains("cs2030 lab 31"));
```

**Review by thamruicong — COMMENTED** [review]
LGTM, merge conflict not resolved

**Comment by leewaichong** [comment]
Good addition of the test cases for code coverage. Small conflict issue in one of the test files. Looks good to merge afterward.

**Review by thamruicong — APPROVED** [review]
Good code coverage, LGTM

**Pull request merged**